### PR TITLE
Allow posting tweets that only contain media (no text)

### PR DESCRIPTION
**Pull request by JoshKeegan**
See #774 

Also removes `IPublishTweetOptionalParameters`, as even text is now optional.

### CODE REVIEW

File: Tweetinvi.Core/Public/Parameters/PublishTweetParameters.cs [hunk 1]
```diff
@@ -14,16 +14,6 @@ public interface IPublishTweetParameters : ICustomRequestParameters
         /// </summary>
         string Text { get; }
```
Comment (linvi) [inline Tweetinvi.Core/Public/Parameters/PublishTweetParameters.cs]: If text becomes not mandatory I think it makes sense to make it `{ get; set; }`

File: Tweetinvi.Core/Public/Parameters/PublishTweetParameters.cs [hunk 2]
```diff
@@ -69,6 +59,11 @@ public interface IPublishTweetParameters : ICustomRequestParameters
         /// </summary>
         List<byte[]> MediaBinaries { get; }
```
Comment (linvi) [inline Tweetinvi.Core/Public/Parameters/PublishTweetParameters.cs]: Removing `PublishTweetOptionalParameters` implies changing the lists settable.

``` c#
public List<long> MediaIds { get; set; } = new List<long>();
public List<IMedia> Medias { get; set; } = new List<IMedia>();
public List<byte[]> MediaBinaries { get; set; } = new List<byte[]>();
```

**Comment by mikemike396** [comment]
Any idea when 5.0 will be released? Thanks

**Comment by linvi** [comment]
Hi, I am very actively working on it for the last 3 weeks but the amount of work required is very significant as it is a total refactor of the library. At the current stage I cannot provide any good ETA.

**Comment by linvi** [comment]
You can follow the evolution on that branch if you are interested:

https://github.com/linvi/tweetinvi/tree/AsyncClient

**Comment by mikemike396** [comment]
Any update on this release? Thanks!

**Comment by mikemike396** [comment]
Awesome! We will give it a try. Thanks again!

**Pull request merged**